### PR TITLE
Change address of dashboard when changing wallets

### PR DESCRIPTION
**Pull request by caxtonacollins**
https://github.com/user-attachments/assets/7aff7db3-14d6-4a5e-a560-f2e29c48978a

## Release Notes

- **New Features**
	- Enhanced user data handling with improved error logging and notifications.
	- Optimized address formatting in the `ProfileCard` component.

- **Bug Fixes**
	- Improved error handling in data fetching functions for better user feedback.

- **Style**
	- Standardized use of single quotes across various components for consistency.
	- Reformatted import statements and JSX structures for improved readability.

- **Chores**
	- Minor adjustments in `tsconfig.json` for JSON formatting compliance.

### CODE REVIEW

**Comment by caxtonacollins** [comment]
@kfastov please review

**Comment by kfastov** [comment]
@caxtonacollins Sorry, was busy on HH, will review today

File: app/provider.tsx [hunk 1]
```diff
@@ -130,7 +133,9 @@ export function Providers({ children }: { children: React.ReactNode }) {
           <NotificationProvider>
             <ThemeProvider theme={theme}>
               <PostHogProvider client={posthog}>
-                <QuestsContextProvider>{children}</QuestsContextProvider>
+                <WalletProvider>
```
Comment (fricoben) [inline app/provider.tsx]: I don't think you need a walletProvider here to do the job, it adds unnecessary complexity. 

What I would do instead is to use starknet react `useAccount()` hook to change URL based on the address you can get from it, you can do that directly on the `[addressOrDomain]/page.tsx` file. 

Here is an example of what i would look like. 

```tsx
export default function Page({ params }: AddressOrDomainProps) {
  // ... existing state declarations ...
  const [connectedAddress, setConnectedAddress] = useState<string | undefined>(); // address set as connected
  const { address } = useAccount(); // current address connected
  
  // Add this useEffect to handle address changes and URL updates
  useEffect(() => {    
    // Skip if the address hasn't changed or isn't defined
    if (!address || address === connectedAddress) return;
    
    // Update connected address state
    setConnectedAddress(address);
    
    // If we're not on the current user's profile, don't redirect
    if (address && addressOrDomain !== address && addressOrDomain !== undefined) {
      return;
    }
    
    // Redirect to the new address profile
    if (address) {
      router.push(`/${address}`);
    }
  }, [address, addressOrDomain, connectedAddress, router]);

  // ... rest of the component ...
}
```

I might have done some mistake here so I'll let you test and adapt but that solution would be way easier than adding a provider (starknet-react is your provider here)

**Review by fricoben — CHANGES_REQUESTED** [review]
You got the overall idea, I propose you something way easier using starknet-react features

**Comment by caxtonacollins** [comment]
@fricoben please review

**Review by fricoben — CHANGES_REQUESTED** [review]
Need to fix conflcits and we're good

**Comment by fricoben** [comment]
@caxtonacollins waiting for your conflicts fix

**Comment by caxtonacollins** [comment]
@fricoben please review, conflicts resolved

**Review by fricoben — APPROVED** [review]

**Pull request merged**